debezium-core/src/main/java/io/debezium/schema/FieldNameSelector.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.schema;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.relational.Column;
+import io.debezium.util.BoundedConcurrentHashMap;
+import io.debezium.util.BoundedConcurrentHashMap.Eviction;
+
+/**
+ * Implementations return names for fields.
+ *
+ * @author Chris Cranford
+ */
+public class FieldNameSelector {
+
+    private final FieldNamer fieldNamer;
+
+    private FieldNameSelector(boolean sanitizeFieldNames) {
+        this.fieldNamer = new FieldNameCache(new FieldNameSanitizer(sanitizeFieldNames, Column::name));
+    }
+
+    public static FieldNameSelector defaultSelector(boolean sanitizeFieldNames) {
```

Comment:
Should `sanitizeFieldNames` really be propagated? Maybe this method can either return `FieldNameSanitizer` which will always sanitize or a lambda that will always return the clumn name?

Reply:
Changed in commit fce5d9a